test/ci/verify-templates.test.js

```diff
@@ -0,0 +1,137 @@
+                const functionsDir = path.join(projectRoot, template,
+                                               'functions');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            it('should have a populated tests directory', (done) => {
+                const functionsDir = path.join(projectRoot, template, 'tests');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            describe('its assets/index.html file', () => {
+                const indexFile = path.join(projectRoot, template,
+                                            'assets', 'index.html');
+
+                it('should exist', (done) => {
+                    fs.access(indexFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should contain <!-- APP_INFO_V2 --> (for new-style index pages)', (done) => {
+                    fs.readFile(indexFile, (err, data) => {
+                        expect(err).toBeFalsy();
+                        if(data.includes('ce-paste-theme.css')) {
+                            expect(data.includes('<!-- APP_INFO_V2 -->'));
+                        }
+                        done();
+                    });
+                });
+            });
+
+            describe('its .env file', () => {
+                const envFile = path.join(projectRoot, template, '.env');
+
+                it('should exist', (done) => {
+                    fs.access(envFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should be parseable', async () => {
+                    const result = parser.parseFile(envFile);
+
+                    await expect(result).resolves.toBeTruthy();
+                });
+
+                it('should have a description for each variable', async () => {
+                    const result = await parser.parseFile(envFile);
+
+                    result.variables.forEach((v) => {
+                        if(!v.description) {
+                            throw new Error(`${v.key} is missing a description`);
+                        }
+                    });
+                });
+
+                it('should have non-configurable TWILIO_*_WEBHOOK_URL variables, if present', async () => {
+                    const result = await parser.parseFile(envFile);
+                    const varRegex = /^TWILIO_.*_WEBHOOK_URL$/;
+
+                    result.variables
```

Comment:
would it make more sense to filter for values matching the regex pattern then check for those configurable values to be falsy? Something like
```
result.variables
  .filter((v) => varRegex.test(v.key))
  .forEach((v) => expect(v.configurable).toBeFalsy())
```

Reply:
You're right, this is cleaner and probably faster.